### PR TITLE
supported the search component for smartphones

### PR DESCRIPTION
**Pull request by adamrocker**
No keywords, no filter condition.
スクリーンショット 2021-09-16 12 58 17

With a keyword and filter conditions.
スクリーンショット 2021-09-16 12 56 46

The search component for a smartphone.
スクリーンショット 2021-09-16 12 57 09

### CODE REVIEW

**Review by yoichiro — CHANGES_REQUESTED** [review]
@adamrocker When setting some conditions and clicking the CLOSE (x) button on the search dialog, these selecting conditions are displayed on the top of the search result page, but actually the search logic is not executed. It is not the behavior users expect, I think. If we remain the close button on the dialog, all changes on the search dialog should be dropped after clicking the close button. Otherwise, we should remove the close button from the search dialog, I think. How about this?

**Comment by adamrocker** [comment]
@yoichiro Thanks for your feedback. I think it would be better for users to be able to close the dialog so, I will reset the search conditions unless a user clicks the close button.

**Comment by yoichiro** [comment]
@adamrocker I agree your idea. Thanks!

File: src/components/catalog/search/CatalogSearch.tsx [hunk 1]
```diff
@@ -62,343 +41,104 @@ class CatalogSearch extends React.Component<
...
-  ) {
-    const value = event.target.value as
-      | IKeyboardHotswapType
-      | IConditionNotSelected;
-    this.props.updateFeatures!(value, ALL_HOTSWAP_TYPE);
+  private isSmallDisplay() {
+    // eslint-disable-next-line no-undef
+    const width = window.innerWidth;
+    return width < 600;
   }
```
Comment (yoichiro) [inline src/components/catalog/search/CatalogSearch.tsx:54]: @adamrocker Should we define this function as a utility function?

Comment (adamrocker) [reply src/components/catalog/search/CatalogSearch.tsx:54]: Could be nice for our future. I cut this method out as a utility function.

**Review by yoichiro — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**